Functions/Import/xASL_imp_CheckImportSettings.m

```diff
@@ -49,7 +49,31 @@
         fprintf('If you want to overwrite, first remove the full subject folder...');
     end
 
+    
+%% Check if folderHierarchy matches with tokenOrdering
+nLeftBrackets = 0;
```

Comment:
This ignores the fact that a bracket can be added as a symbol as well.
In 90% of cases it's nice to report as a possible error. But in 10% of cases, this is a valid entry and shouldn't result in an error! Make this a warning only and a conditional one!

The counting of tokens is 100% wrong. There is no rule that the number of tokens defined has to match the number of tokens. We can make a warning if bracket number is smaller than the number of tokens. But inequality should be only reported as suspicious, never as an error!

Reply:
The only rule that I enforced here, is that the number of tokens in tokenOrdering equals the number of tokens in folderHierarchy, which should always be the case, otherwise it will crash further down the code. If we only issue a warning here, the code continuous and can incorrectly use the wrong tokenOrdering token with the wrong folder in the folderHierarchy. This can then produce a result that doesn't crash the pipeline, but is incorrect and takes time to bugfix. So I would keep this check.

Reply:
Yes, I'm fine with checking the number of tokens. BUT :) The number of brackets doesn't have to equal the number of tokens. There are other uses of brackets than for tokens...

Reply:
Nono. I mean that in regexp, there are other uses of the bracket character that is not defining tokens!

Reply:
That's what I'm saying. The folderHierarchy regexp doesn't allow () outside the use of defining tokens. Otherwise, we don't know if () is for either/or for defining tokens, right?

So when the number of brackets is not according to the number of tokens, I now added a hint to the error, for the user to understand that you can only use () for tokens.

Reply:
See my last edit, does this clarify it?